@here/harp-datasource-protocol/lib/StyleSetEvaluator.ts

```diff
@@ -372,6 +377,7 @@ export class StyleSetEvaluator {
 
         this.changeLayer(previousLayer); // restore the layer
         this.changeGeometryType(previousGeometryType); // restore the geometryType
+        this.changeZoomLevel(previousZoomLevel);
```

Comment:
This is actually question to #1096 ...
Is it still necessary to restore previous zoom/geomType/layer when we don't have nested styles ?

Reply:
Yes, now that the nested stylesets are gone the StyleSetEvaluator can be greatly simplified.
We could remove a lot of code and like the context handling (m_layer, m_geometrytype and m_zoomlevel and the change<Context> methods). I was trying to keep the change small to make it easier to review but of course, if we want we could also refactor that code in the context of this PR.